crmd/lrm.c

```diff
@@ -694,13 +687,69 @@ append_secure_list(lrmd_event_data_t *op, xmlNode *metadata, xmlNode * update, c
     free(list);
 }
 
+regex_t *version_format_regex = NULL;
+
+static gboolean
+valid_version_format(const char *version)
+{
+    if (version == NULL) {
+        return FALSE;
+    }
+
+    if (version_format_regex == NULL) {
+        const char *regex_string = "^[[:digit:]]+([.][[:digit:]]+)*$";
+        version_format_regex = calloc(1, sizeof(regex_t));
+        regcomp(version_format_regex, regex_string, REG_EXTENDED);
+    }
+
+    return regexec(version_format_regex, version, 0, NULL, 0) == 0;
+}
+
+static const char*
+get_ra_version(lrmd_rsc_info_t *rsc, xmlNode *metadata)
+{
+    static const char *default_version = "0.1";
+    const char *version = NULL;
+    xmlNode *version_xml = NULL;
+
+    if (!metadata) {
+        return default_version;
+    }
+
+    version_xml = get_xpath_object("//resource-agent/@version", metadata, LOG_TRACE);
+    if (!version_xml) {
+        if (rsc) {
+            crm_info("Resource agent version for %s (%s:%s:%s) doesn't stated",
+                      rsc->id, rsc->class, rsc->provider, rsc->type);
```

Comment:
Let's lower this to debug -- a lot of existing agents don't specify a version, and users might get concerned if this starts repeating in their logs. Also, let's rephrase as:

Resource agent %s:%s:%s (for %s) doesn't specify a version number